Libraries/Azure.psm1

```diff
@@ -213,57 +205,179 @@ Function Validate-SubscriptionUsage($RGXMLData, $Location, $OverrideVMSize, $Sto
+	$regionStorageMapping = [xml](Get-Content .\XML\RegionAndStorageAccounts.xml)
+	if ($StorageAccount) {
+		if ($StorageAccount -imatch "^ExistingStorage_Standard") {
+			$StorageAccountName = $regionStorageMapping.AllRegions.$RegionName.StandardStorage
+		}
+		elseif ($StorageAccount -imatch "^ExistingStorage_Premium")
+		{
+			$StorageAccountName = $regionStorageMapping.AllRegions.$RegionName.PremiumStorage
+		}
+	}
+	else {
+		$StorageAccountName = $regionStorageMapping.AllRegions.$RegionName.StandardStorage
+	}
+	Write-LogInfo "Selected : $StorageAccountName"
+	return $StorageAccountName
+}
+
 Function Change-StorageAccountType($TestCaseData, [string]$Location, $GlobalConfig, $OsVHD) {
-    $storageAccount = $GlobalConfig.Global.Azure.Subscription.ARMStorageAccount
-    $changedSC = ""
-    $copyVHD = $false
-    if ($TestCaseData.AdditionalHWConfig.StorageAccountType -and $TestCaseData.AdditionalHWConfig.StorageAccountType.Contains("Premium")) {
-        # if SC is ExistingStorage_Standard, switch to ExistingStorage_Premium
-        if ($storageAccount -imatch "ExistingStorage_Standard") {
-            $changedSC = Get-StorageAccountFromRegion -Region $Location -StorageAccount "ExistingStorage_Premium"
-            # if it is OsVHD format, need copy VHD from standard to premium storage account
-            if ($OsVHD) {
-                $copyVHD = $true
-            }
-        # if SC is NewStorage_Standard, switch to NewStorage_Premium
-        } elseif ($storageAccount -imatch "NewStorage_Standard") {
-            $changedSC = "NewStorage_Premium_LRS"
-        } elseif (($storageAccount -imatch "NewStorage_Premium") -or ($storageAccount -imatch "ExistingStorage_Premium")) {
-            $changedSC = ""
-        } else {
-            $current_sc = Get-StorageAccountFromRegion -Region $Location -StorageAccount "ExistingStorage_Standard"
-            if ($current_sc -eq $storageAccount) {
-                $changedSC = Get-StorageAccountFromRegion -Region $Location -StorageAccount "ExistingStorage_Premium"
-                if ($OsVHD) {
-                    $copyVHD = $true
-                }
-            } else {
-                $storageAccountType = (Get-AzStorageAccount | Where-Object {$_.StorageAccountName -eq $storageAccount}).Sku.Tier.ToString()
-                if ($storageAccountType -inotmatch "premium") {
-                    Write-LogErr "Provided storage account is not premium type, this case $($TestCaseData.testName) need run under premium type of storage account."
-                    Throw "Case $($TestCaseData.testName) need run under premium type of storage account."
-                }
-            }
-        }
-    }
+	# From PrepareTestEnvironment(), $GlobalConfig.Global.Azure.Subscription.ARMStorageAccount has already been existing Storage Account Name
+	$storageAccount = $GlobalConfig.Global.Azure.Subscription.ARMStorageAccount
+	$changedSC = $storageAccount
```

Comment:
Can you change the variable name 'changeSC' more human recognizable? Or easy to read? You have changeSC and current_sc, sourceContainer. It's hard to read with them.

Reply:
it's just previous defined variable names, by previous author.  Now updated, please take a look.